src/cfg_nop_elim.ml

```diff
@@ -53,17 +51,20 @@ let nop_transitions_closed cfgfunbody =
 
 (* [replace_succ nop_succs s] donne le nouveau nom du nœud [s], en utilisant la
    liste [nop_succs] (telle que renvoyée par [nop_transitions_closed]). *)
-let replace_succ nop_succs s =
-   match List.assoc_opt s nop_succs with
-   | None -> s
-   | Some s' -> s'
-
+let rec replace_succ nop_succs s = match nop_succs with
+   |(x,y)::q when x=s -> y
+   |(x,y)::q -> replace_succ q s
+   | _ -> s
 (* [replace_succs nop_succs n] remplace le nœud [n] par un nœud équivalent où on
    a remplacé les successeurs, en utilisant la liste [nop_succs]. *)
-let replace_succs nop_succs (n: cfg_node) =
-   match n with
-   | Cnop(s) -> Cnop(replace_succ nop_succs s)
-   | _ -> n
+let replace_succs nop_succs (n: cfg_node) = match n with 
+   | Cassign (str, e, s) -> let s' = replace_succ nop_succs s in Cassign (str, e, s')
+   | Creturn e -> n
+   | Cprint (e, s) -> let s' = replace_succ nop_succs s in Cprint (e, s')
```

Comment:
remove Cprint line, replace with

` 
| Ccall (str, e, s) -> let s' = replace_succ nop_succs s in Ccall (str, e, s')
| Cstore(e1, e2, v, s) -> let s' = replace_succ nop_succs s in Cstore(e1, e2, v, s')
  `